Pandas ix deprecation => Switch to loc and iloc

jklenzing opened the pull request:
This is largely because I want a review of the approach so far on addressing issue #16.  I've updated the code in getitem, but not setitem.  I am testing against the examples listed in lines 354-365 of _instrument.py.  It works for all of these except the last one:

Slicing by name and row/date
inst[datetime1:datetime1, 'name1':'name2']

Specifically, it's the 'name1':'name2' slice that is giving trouble.  I've not found a lot of information about how to structure this, and I'm not certain how useful a slice of variable names is.  Is there a reason to continue to support this particular syntax?

jklenzing commented:
Removing review requests until it passes travis tests

jklenzing commented:
Found a better implementation, which means we should be able to keep all functionality listed in the example lines.  Tested the code using test_instruments.py locally.  Now let's see what Travis has to say.

jklenzing commented:
Using the syntax recommended here: http://pandas-docs.github.io/pandas-docs-travis/indexing.html#ix-indexer-is-deprecated

Defaulting to using .loc as opposed to .iloc since it works for both datetime and string objects.

jklenzing commented:
Failing to pass travis because some of the data cleaning routines pass through a numpy array of integers, which I fail to check.  Working on a solution.

jklenzing commented:
So far, I've eliminated pandas.ix in _instrument.py, _constellation.py, and utils.py.  The only remaining instances I see are in _files.py.  The 3.6 version looks good, but 2.7 is failing here (and in #126).

From the travis log:  "No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received"

Thoughts?  This occurs during the pyglow build.

jklenzing commented:
This should do it.  I've replaced every instance of .ix I could find, with the exception of those that are commented out and 1 line in the tutorial section (need to download cosmic data and see how I would construct this now).

rstoneback commented:
pyglow plus the travis cache system frequently causes issues.
Generally there isn’t much we can do once the pyglow clone and travis system get into certain funks.
Sometimes restarting the build will work, but not in this case.

I’ve modified the travis.yml in another branch to change rm -r to rm -rf, unfortunately because of the way travis sets things up it becomes impossible to get rid of the directory and then git won’t download a new clone.

It if makes you feel better one of my other branches in is in the exact same state.



On Nov 30, 2018, at 3:53 AM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:


So far, I've eliminated pandas.ix in _instrument.py, _constellation.py, and utils.py. The only remaining instances I see are in _files.py. The 3.6 version looks good, but 2.7 is failing here (and in #126<https://github.com/rstoneback/pysat/pull/126>).

From the travis log: "No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received"

Thoughts? This occurs during the pyglow build.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/125#issuecomment-443151072>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeYYGFQTfsh5NCgSW8887NaJBdpfgks5u0QAQgaJpZM4Y15nx>.

jklenzing commented:
Yeah, it just did it again.

jklenzing commented:
_files.py is the biggest problem I see in the testing.  Reverting to ix for this file so we can run the others through travis and discuss.

aburrell commented:
@jklenzing you seem to be making a lot of changes to this still.  Do you want a review now, or do you want to wait until this pull request is stable?

jklenzing commented on pysat/_instrument.py:
```diff
@@ -342,71 +342,83 @@ def __init__(self, platform=None, name=None, tag=None, sat_id=None,
+                    idx = self.data.index[key[0]]
+                elif isinstance(key[0],slice) and isinstance(key[0].start,int):
+                    idx = self.data.index[key[0]]
+                elif isinstance(key[0],ndarray):
```
Updating line 380 to

elif isinstance(key[0],ndarray) and isinstance(key[0,0],np.int64):

jklenzing replied:
Also added a try-except here to send an error message in case all this fails

jklenzing commented on pysat/_files.py:
```diff
@@ -405,31 +408,33 @@ def __getitem__(self, key):
             else:
                 # not a datetime
                 return out
+        elif isinstance(key, int) or isinstance(key, np.int64):
```
updated

jklenzing commented:
@rstoneback, I've updated to use duck typing to sort out the possible inputs.  It works, but I have a nested try-except structure to sort out three possible options.  Not sure if there is a more compact way to do this.

Check out lines 373-390 in _instrument.py as an example of what I mean.

jklenzing commented:
@aburrell, @rstoneback, this is stable and ready for review.  Addresses #16, #63, and #129.  The travis failures are the usual issues with downloads.

jklenzing commented on .travis.yml:
```diff
@@ -48,37 +49,21 @@ install:
-  - if [[ -d "$PYGLOW_DIR" && -e "$PYGLOW_DIR/setup.py" && "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
-      cd $PYGLOW_DIR;
-      python setup.py install;
+  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
```
This was based on one of @rstoneback's updates to travis.yml for one of the other pull requests

jklenzing commented on pysat/tests/test_instrument.py:
```diff
@@ -503,24 +585,30 @@ def teardown(self):
         self.rawInst.load(fid=3)
-        assert ( (self.testInst.index[0] == self.rawInst.index[0] - pds.DateOffset(minutes=5)) & 
-                (self.testInst.index[-1] == self.rawInst.index[-1] + pds.DateOffset(minutes=5)) )
+        assert ( (self.testInst.index[0] ==
```
fixed

jklenzing commented on pysat/tests/test_instrument.py:
```diff
@@ -530,32 +618,39 @@ def test_fid_data_padding_prev(self):
         #   self.testInst.index[-1],  self.rawInst.index[-1] + pds.DateOffset(minutes=5))
-        assert ( (self.testInst.index[0] == self.rawInst.index[0] - pds.DateOffset(minutes=5)) & 
-                (self.testInst.index[-1] == self.rawInst.index[-1] + pds.DateOffset(minutes=5)) )
+        assert ( (self.testInst.index[0] ==
```
fixed

jklenzing commented on pysat/tests/test_utils.py:
```diff
@@ -35,7 +35,7 @@ def test_getyrdoy_leap_year():
     date = pds.datetime(2008,12,31)
     yr, doy = pysat.utils.getyrdoy(date)
-    assert ((yr == 2008) & (doy == 366)) 
+    assert ((yr == 2008) & (doy == 366))
```
not seeing this in test_utils.py.

aburrell replied:
Probably a good call to just search for all the assert statements, that's where I was seeing this happening.

jklenzing replied:
thorough search and pep8 update to this file

jklenzing commented on pysat/tests/test_instrument.py:
```diff
@@ -177,191 +200,250 @@ def test_instrument_init(self):
+
     def test_setting_data_by_name_with_meta(self):
         self.testInst.load(2009,1)
         self.testInst['doubleMLT'] = {'data':2.*self.testInst['mlt'],
```
updated

jklenzing commented on pysat/_instrument.py:
```diff
@@ -442,40 +460,53 @@ def __getitem_xarray__(self, key):
+                try:
+                    # Assume key[0] is integer (including list or slice)
+                    self.data.loc[self.data.index[key[0]], key[1]] = new
+                except:
```
reorganized to use KeyError and TypeError in the selection criteria

jklenzing commented:
Fixes for issues #16, #63, and #129.  Adds a unit test to check for issue #63 in future code (pandas only).

jklenzing commented:
Waiting on merge of #155 and #143.

jklenzing commented:
Summary of Changes:
- Removal of pandas ix notation throughout, replaced with iloc and loc (#16)
- Added unit tests for more types of data slicing, including the bug outlined in #63
- Updated `.reindex_axis` to `.reindex` (#129)
- Added `sort=True` to usage of `pandas.concat` to maintain current behaviour (#152)
- Deleted commented functions in `test_meta.py` (@rstoneback, is this okay?)
- various PEP8 scrubs throughout

jklenzing commented:
Apologies for the lengthy diff files.  A lot of this branch predates #142.

jklenzing commented:
Also up for review, tackling #158 as part of this pull is likely trivial, since it just involves adding a few keywords to maintain behaviour.

jklenzing commented:
Hmm.  JRO download fail in the pr side.  Restarting trvais jobs..

jklenzing commented:
OK, Travis CI passes!  Coverage should go up in #155

jklenzing commented:
Adding unit tests...
